src/mlpack/methods/dbscan/random_point_selection.hpp

```diff
@@ -45,14 +53,20 @@ class RandomPointSelection
         ++found;
 
       if (found > index)
+      {
+        unvisited[i].flip(); // Set unvisited point to visited point
         return i;
+      }
     }
-
     return 0; // Not sure if it is possible to get here.
   }
-};
 
+ private:
+  // Bitmask for unvisited points, If true, mean unvisited.
+  boost::dynamic_bitset<> unvisited;
+};
+  
 } // namespace dbscan
 } // namespace mlpack
 
-#endif
+#endif
```

Comment:
This is a really tiny issue, but can you add the newline back to the end of the file please?  Otherwise `cat random_point_selection.hpp` will give weird results when it prints the prompt right after the `#endif`. :)